stamp.sh

```diff
@@ -7,3 +7,5 @@ CURRENT_TAG=$(git tag -l --points-at HEAD | head -n1)
 CURRENT_COMMIT=$(git rev-parse HEAD)
 
 echo "STABLE_VERSION ${CURRENT_TAG:-$CURRENT_COMMIT}"
+# rules_nodejs expects to read from volatile-status.txt
+echo "BUILD_SCM_VERSION ${CURRENT_TAG:-$CURRENT_COMMIT}"
```

Comment:
I've found it valuable to add a `-dirty` suffix when user changes happen. This saved me trying to debug someone's issue which would have been impossible without it

Reply:
Agreed, for Angular we use a `.with-local-changes` suffix in a non-pristine clone
https://github.com/angular/angular/blob/master/tools/bazel_stamp_vars.js#L52-L53

But of course that change should be made for all the stamping here, not just npm, so I don't think it should go in this PR. @philwo do you want a separate PR for that?